src/toil/provisioners/gceProvisioner.py

```diff
@@ -0,0 +1,915 @@
+
+    def remainingBillingInterval(self, node):
+        #TODO - does this exist in GCE?
+        return 1 #awsRemainingBillingInterval(node)
+
+    def terminateNodes(self, nodes):
+        nodeNames = [n.name for n in nodes]
+        instances = self._getNodesInCluster()
+        instancesToKill = [i for i in instances if i.name in nodeNames]
+        self._terminateInstances(instancesToKill)
+
+    def addNodes(self, nodeType, numNodes, preemptable):
+        # If keys are rsynced, then the mesos-slave needs to be started after the keys have been
+        # transferred. The waitForKey.sh script loops on the new VM until it finds the keyPath file, then it starts the
+        # mesos-slave. If there are multiple keys to be transferred, then the last one to be transferred must be
+        # set to keyPath.
+        keyPath = ''
+        entryPoint = 'mesos-slave'
+        botoExists = False
+        if self.botoPath is not None and os.path.exists(self.botoPath):
+            entryPoint = "waitForKey.sh"
+            keyPath = nodeBotoPath
+            botoExists = True
+        elif self.config and self.config.sseKey:
+            entryPoint = "waitForKey.sh"
+            keyPath = self.config.sseKey
+
+        workerData = dict(role='worker',
+                          dockerImage=applianceSelf(),
+                          entrypoint=entryPoint,
+                          sshKey=self.masterPublicKey,
+                          dockerArgs=workerDockerArgs.format(ip=self.leaderIP, preemptable=preemptable, keyPath=keyPath))
+
+        #kwargs["subnet_id"] = self.subnetID if self.subnetID else self._getClusterInstance(self.instanceMetaData).subnet_id
+
+        userData = self.gceUserDataWorker.format(**workerData)
+        metadata = {'items': [{'key': 'user-data', 'value': userData}]}
+
+        imageType = 'coreos-stable'
+        sa_scopes = [{'scopes': ['compute', 'storage-full']}]
+
+        # TODO:
+        #  - bug in gce.py for ex_create_multiple_nodes (erroneously, doesn't allow image and disk to specified)
+        #  - ex_create_multiple_nodes is limited to 1000 nodes
+        #    - use a different function
+        #    - or write a loop over the rest of this function, with 1K nodes max on each iteration
+
+        if not preemptable:
+            logger.info('Launching %s non-preemptable nodes', numNodes)
+        else:
+            logger.info('Launching %s preemptable nodes', numNodes)
+
+        disk = {}
+        disk['initializeParams'] = {
+            'sourceImage': bytes('https://www.googleapis.com/compute/v1/projects/coreos-cloud/global/images/coreos-stable-1576-4-0-v20171206'),
+            'diskSizeGb' : self.nodeStorage }
+        disk.update({'boot': True,
+             #'type': 'bytes('zones/us-central1-a/diskTypes/local-ssd'), #'PERSISTANT'
+        #     'mode': 'READ_WRITE',
+        #     'deviceName': clusterName,
+             'autoDelete': True })
+        #instancesLaunched = driver.ex_create_multiple_nodes(
```

Comment:
Delete this.

Reply:
Not yet. The reason is described in a comment above.